.github/workflows/bump-version.yml

```diff
@@ -1,44 +1,106 @@
+name: Bump Version
+
 on:
-    pull_request:
-      branches:
-        - main
-      types: [closed]
-  
+  pull_request:
+    types: [closed]
+    branches: [main]
+
 jobs:
-    bump-version:
-      runs-on: ubuntu-latest
-  
-      env:
-        GH_PUSH_TOKEN: ${{ secrets.GH_PUSH_TOKEN }}
-  
-      steps:
-  
-        - name: Checkout
-          uses: actions/checkout@v4
-          if: env.GH_PUSH_TOKEN != ''
-          with:
-            token: ${{ env.GH_PUSH_TOKEN }}
-  
-        - name: configure git
-          if: env.GH_PUSH_TOKEN != ''
-          run: |
-            git remote set-url origin https://itsalaidbacklife@github.com/cuttle-cards/cuttle
-            git config --global user.name "Cuttle Continuous Integration"
-            git config --global user.email "itsalaidbacklife@gmail.com"
-  
-  
-        - name: bump patch version
-          if: contains(github.event.pull_request.labels.*.name, 'version-patch') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:patch
-  
-        - name: bump minor version
-          if: contains(github.event.pull_request.labels.*.name, 'version-minor') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:minor
-  
-        - name: bump major version
-          if: contains(github.event.pull_request.labels.*.name, 'version-major') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:major
-  
-        - name: push
-          if: env.GH_PUSH_TOKEN != ''
-          run: git push origin main
+  check-token:
+    runs-on: ubuntu-latest
+    outputs:
+      GH_PUSH_TOKEN: ${{ steps.check.outputs.GH_PUSH_TOKEN }}
+    steps:
+      - name: Check GH_PUSH_TOKEN
+        id: check
+        run: |
+          if [[ -z "${{ secrets.GH_PUSH_TOKEN }}" ]]; then
+            echo "GH_PUSH_TOKEN is not set. Exiting."
+            exit 1
+          fi
+          echo "::set-output name=GH_PUSH_TOKEN::${{ secrets.GH_PUSH_TOKEN }}"
+        shell: bash
+
+  check-merged:
```

Comment:
3. **Check Merge Status:** The `check-merged` job runs next, contingent on the successful completion of `check-token`. It verifies if the pull request was merged. If it wasn't, the job fails and the workflow terminates. If the pull request was merged, the job outputs the merge status for subsequent jobs.

This is the fix for the above callout-- if the branch was closed AND ALSO MERGED, then we can continue.

Reply:
Similarly to above, why is erroring when the PR was manually closed better than no-op succeeding?

Reply:
I've switched this to noop each consecutive job is the token isn't set.